### PR TITLE
GAWB-1405 select2 wip

### PR DESCRIPTION
**Pull request by skwalker**
~~started reimplementing tagging with select2; it is mostly working except for~~
~~1. I can't get defaults to work with autocomplete (as in, if you've already saved tags, we would like to show them as tags as well as let you autocomplete for new things). current options I've found are EITHER only see tags / don't autocomplete, OR see X where tag name should be and autocomplete works.. it's really weird, when you autocomplete for a tag that should be there the whole thing just disappears. it's greeeaaaattttt. I think http://stackoverflow.com/questions/30316586/select2-4-0-0-initial-value-with-ajax#30328989.~~
~~2. needs a lot of ux improvements (talk to Isaac).. I tried to add a sass file and do some minor fixes but they weren't showing up idk why.~~

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/cljs/main/broadfcui/page/workspace/summary/tab.cljs [hunk 1]
```diff
@@ -127,14 +128,14 @@
...
-                                           {:on-success (fn [namespace name]
-                                                          (swap! state dissoc :cloning?)
-                                                          (on-clone (str namespace ":" name)))
-                                            :workspace-id workspace-id
-                                            :description description
-                                            :is-protected? isProtected
-                                            :billing-projects billing-projects}])}])
+                                           [WorkspaceCloner
+                                            {:on-success (fn [namespace name]
+                                                           (swap! state dissoc :cloning?)
```
Comment (skwalker) [inline src/cljs/main/broadfcui/page/workspace/summary/tab.cljs:133]: might've changed spacing incorrectly (I just re-downloaded intellij and I'm not convinced all the settings are right, sorry)

**Comment by zarsky-broad** [comment]
I'm hopping on this to do some cleanup.

**Comment by zarsky-broad** [comment]
All set, ready for review.

File: src/cljs/main/broadfcui/page/workspace/details.cljs [hunk 2]
```diff
@@ -78,6 +78,10 @@
                                 (filter (fn [[k _]]
                                           (.startsWith (name k) "library:")))
                                 (into {}))
+        tags (->> attributes
+                  (filter (fn [[k _]]
+                            (.startsWith (name k) "tag:")))
+                  (into {}))
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/page/workspace/details.cljs]: This whole thing could just be `(utils/filter-keys #(.startsWith (name %) "tag:") attributes)`

File: src/cljs/main/broadfcui/page/workspace/summary/tab.cljs [hunk 3]
```diff
@@ -151,12 +152,13 @@
...
   (let [{:keys [owners]
-         {:keys [createdBy createdDate bucketName description workspace-attributes library-attributes realm]} :workspace} workspace
+         {:keys [createdBy createdDate bucketName description tags workspace-attributes library-attributes realm]} :workspace} workspace
         realm-name (:realmName realm)
         render-detail-box (fn [order title & children]
                             [:div {:style {:flexBasis "50%" :order order}}
                              (style/create-section-header title)
-                             children])]
+                             children])
+        processed-tags (flatten (map #(:items %) (vals tags)))]
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/page/workspace/summary/tab.cljs]: `#(:items %)` -> `:items`

**Review by MattPutnam — APPROVED** [review]
👍 looks good, two minor cleanup things

**Review by davidangb — APPROVED** [review]
works well for me in practice, though I haven't done comprehensive tests. Code looks good modulo Matt's small comments. 👍

**Comment by davidangb** [comment]
Moving from `select2.full` to `select2` doesn't seem to have any impact in tags or in method config importing, which is the other place we use select2 ... but I'm not sure all the things to test. Would this have any impact on our other usage of select2?

**Comment by zarsky-broad** [comment]
@davidangb Nope, the move to full was done just for this PR because Sarah thought it was needed, but it's not. Other than that, I've just cleaned up the syntax of the import.

**Pull request merged**